packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -87,104 +87,146 @@ class LoadTestDataStoreModel {
             throw new Error("taskmanger not available");
         }
 
-        if(reset) {
-            await LoadTestDataStoreModel.waitForCatchup(runtime);
-            runDir.set(startTimeKey,Date.now());
-            runDir.delete(taskTimeKey);
-            counter.increment(-1 * counter.value);
-        }
-
-        return new LoadTestDataStoreModel(
+        const dataModel =  new LoadTestDataStoreModel(
+            root,
             config,
             runtime,
             taskmanager,
             runDir,
             counter,
         );
+
+        if(reset) {
+            await LoadTestDataStoreModel.waitForCatchup(runtime);
+            runDir.set(startTimeKey,Date.now());
+            runDir.delete(taskTimeKey);
+            counter.increment(-1 * counter.value);
+            const partnerCounter = await dataModel.getPartnerCounter();
+            if(partnerCounter !== undefined && partnerCounter.value > 0) {
+                partnerCounter.increment(-1 * partnerCounter.value);
+            }
+        }
+        return dataModel;
     }
 
     private readonly taskId: string;
+    private readonly partnerId: number;
     private taskStartTime: number =0;
 
     private constructor(
+        private readonly root: ISharedDirectory,
         private readonly config: IRunConfig,
         private readonly runtime: IFluidDataStoreRuntime,
         private readonly taskManager: ITaskManager,
         private readonly dir: IDirectory,
         public readonly counter: ISharedCounter,
     ) {
+        const halfClients = Math.floor(this.config.testConfig.numClients / 2);
         // The runners are paired up and each pair shares a single taskId
-        this.taskId = `op_sender${Math.floor(config.runId / 2)}`;
+        this.taskId = `op_sender${config.runId % halfClients}`;
+        this.partnerId = this.config.runId < halfClients
+            ? this.config.runId + halfClients
+            : this.config.runId % halfClients;
```

Comment:
```suggestion
            : this.config.runId - halfClients;
```

I think the symmetry of +/- makes this a little clearer what's going on.  I think you could also do `this.partnerId = (this.config.runId + halfClients) % this.config.testConfig.numClients

Reply:
feel free to ignore for now though :)

Reply:
yeah. i had it that way, but if numClients is odd % will behave better